scheduler/test/cook/test/mesos/monitor.clj

```diff
@@ -265,4 +266,18 @@
     (is (= 1 (counter ["total" "users"])))
     (is (= 0 (counter ["starved" "users"])))
     (is (= 0 (counter ["hungry" "users"])))
-    (is (= 1 (counter ["satisfied" "users"])))))
\ No newline at end of file
+    (is (= 1 (counter ["satisfied" "users"])))))
+
+(deftest test-start-collecting-stats
+  (setup :config {:metrics {:user-metrics-interval-seconds 1}})
+  (let [period (atom nil)
+        millis-between (fn [a b]
+                         (-> a
+                             ^Interval (time/interval b)
+                             .toDuration
+                             .getMillis))]
+    (with-redefs [chime/chime-at (fn [p _ _] (reset! period p))]
```

Comment:
Does `with-redefs` of `datomic/conn` work?

Reply:
I'm not sure. Why do you want to, though? Using the in-memory datomic connection seems to work really well.

Reply:
We spoke offline as we want to see how it feels to use this pattern.

Reply:
See `test-redef-config`